plugins/plugins-api/src/main/java/org/codice/compliance/saml/plugin/IdpPostResponse.java

```diff
@@ -133,22 +133,19 @@ private void parseAndSetFormValues(Node responseForm) {
      *
      * And "hidden" means both the SAMLResponse and RelayState MUST be placed in "hidden" form controls
      */
-    // SAMLResponse portion
-    if (samlResponseForm != null) {
-      if (isNotEmpty(samlResponseForm.value())) {
-        samlResponse = samlResponseForm.value();
-      } else if (isNotEmpty(samlResponseForm.attributes().get(VALUE))) {
-        samlResponse = samlResponseForm.attributes().get(VALUE);
-      }
-    }
 
-    // RelayState portion
-    if (relayStateForm != null) {
-      if (isNotEmpty(relayStateForm.value())) {
-        relayState = relayStateForm.value();
-      } else if (isNotEmpty(relayStateForm.attributes().get(VALUE))) {
-        relayState = relayStateForm.attributes().get(VALUE);
+    samlResponse = extractValue(samlResponseForm);
+    relayState = extractValue(relayStateForm);
+  }
+
+  private String extractValue(Node node) {
+    if (node != null) {
+      if (isNotEmpty(node.value())) {
+        return node.value();
+      } else if (isNotEmpty(node.attributes().get(VALUE))) {
```

Comment:
I forgot to add a null check on that `else if` 😀. A NPE could be thrown if `attributes()` is null. A possible refactor could be:
```
private String extractValue(Node node) {
  if (node != null) {
    if (isNotEmpty(node.value())) {
      return node.value();
    } else if (node.attributes() != null && isNotEmpty(node.attributes().get(VALUE))) {
      return node.attributes().get(VALUE);
    }
    return null;
}
```

Reply:
Couldn't you just do this?
```
private String extractValue(Node node) {
  if (node != null) {
    if (isNotEmpty(node.value())) {
      return node.value();
    } else {
      return node.attributes()?.get(VALUE);
    }
}
```
The else will just return null here if `attributes()` is null and will return null if there are no values associated with the `VALUE` key in `attributes().get(VALUE)`.

Reply:
Just kidding you can't do that because this is Java, but just for fun you could even simplify it further by using the Elvis operator (if this was Kotlin)
```
private String extractValue(Node node) {
  return node?.value() ?: node.attributes()?.get(VALUE)
}
```

Reply:
Since this is Java, sadly, I would suggest this:
```java
  private String extractValue(Node node) {
    if (node == null) {
      return null;
    }

    if (isNotEmpty(node.value())) {
      return node.value();
    }

    if (node.attributes() == null) {
      return null;
    }
    
    if (isNotEmpty(node.attributes().get(VALUE))) {
      return node.attributes().get(VALUE);
    }
    
    return null;
  }
```